tsdb/block.go

```diff
@@ -71,11 +73,9 @@ type IndexReader interface {
 	// LabelValues returns possible label values which may not be sorted.
 	LabelValues(ctx context.Context, name string, matchers ...*labels.Matcher) ([]string, error)
 
-	// Postings returns the postings list iterator for the label pairs.
-	// The Postings here contain the offsets to the series inside the index.
-	// Found IDs are not strictly required to point to a valid Series, e.g.
-	// during background garbage collections.
-	Postings(ctx context.Context, name string, values ...string) (index.Postings, error)
+	// PostingsForMatcher returns a sorted iterator over postings having a label matching the provided label matcher.
+	// If no postings are found having a label with the correct name and matching value, an empty iterator is returned.
+	PostingsForMatcher(ctx context.Context, m *labels.Matcher) index.Postings
```

Comment:
Any reason why this is not part of the PostingsReader interface?

Reply:
See the previous [discussion](https://github.com/prometheus/prometheus/pull/13620#discussion_r1497792037) :)